VimeoNetworking/VimeoNetworking/Request.swift

```diff
@@ -223,14 +167,37 @@ public struct Request<ModelType: MappableResponse>
     
     internal func associatedPageRequest(newPath newPath: String) -> Request<ModelType>
     {
+        // Since page response paging paths bake the paging parameters into the path (even if we originally provide them in the body),
+        // strip any parameters now included in the path from the original body parameters.
+        
+        let parameters = self.dynamicType.parametersByRemovingQueryParameters(inPath: newPath, parameters: self.parameters)
+        
         return Request(method: self.method,
                        path: newPath,
-                       parameters: self.additionalParameters,
-                       page: nil,
-                       itemsPerPage: nil,
+                       parameters: parameters,
                        modelKeyPath: self.modelKeyPath,
                        cacheFetchPolicy: self.cacheFetchPolicy,
                        shouldCacheResponse: self.shouldCacheResponse,
                        retryPolicy: self.retryPolicy)
     }
-}
\ No newline at end of file
+    
+    private static func parametersByRemovingQueryParameters(inPath path: String, parameters: AnyObject?) -> AnyObject?
+    {
+        guard var parametersDictionary = parameters as? VimeoClient.RequestDictionary else
```

Comment:
I'm a little confused, it seems we're doing a nil check as well as a type check here. But we don't do anything with the `parameters` argument below. So can we just have the argument type match the type we're checking for here?